packages/trader/src/AppV2/Containers/Notifications/notifications.tsx

```diff
@@ -1,14 +1,47 @@
-import { useNotifications, NotificationBanners } from '@deriv-com/quill-ui';
 import React from 'react';
+import { observer, useStore } from '@deriv/stores';
+import { useNotifications, NotificationBanners } from '@deriv-com/quill-ui';
+import { StandaloneFlagCheckeredFillIcon } from '@deriv/quill-icons';
+import { routes } from '@deriv/shared';
+import { useLocation } from 'react-router-dom';
+
+const Notifications = observer(() => {
+    const { addBanner, banners, removeBanner } = useNotifications();
+    const { portfolio } = useStore();
+    const { setAddNotificationBannerCallback } = portfolio;
+    const { pathname } = useLocation();
+
+    React.useEffect(() => {
+        if (pathname === routes.trade) return;
+        banners.forEach(({ type, id }) => {
+            if (!type) {
+                // Sell notifications have type and Purchase ones do not.
+                removeBanner(id);
+            }
+        });
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [banners, pathname]);
+
+    React.useEffect(() => {
+        const addNotificationBannerCallback = (params: Parameters<typeof addBanner>[0], result: string) =>
+            addBanner({
+                icon: <StandaloneFlagCheckeredFillIcon iconSize='sm' className={`trade-notification--${result}`} />,
+                ...params,
+            });
 
-const Notifications = () => {
-    const { banners, removeBanner } = useNotifications();
+        setAddNotificationBannerCallback(addNotificationBannerCallback);
+        // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
can we call setAddNotificationBannerCallback with undefined on unmount to do clean up and avoid memory leak

Reply:
Yes, it would be better, thank you for noticing)

Reply:
Done, hence resolve the comment